plugin-server/src/kafka/batch-consumer.ts

```diff
@@ -0,0 +1,192 @@
+import { GlobalConfig, Message } from 'node-rdkafka'
+import { exponentialBuckets, Histogram } from 'prom-client'
+
+import { status } from '../utils/status'
+import { createAdminClient, ensureTopicExists } from './admin'
+import {
+    commitOffsetsForMessages,
+    consumeMessages,
+    createKafkaConsumer,
+    disconnectConsumer,
+    instrumentConsumerMetrics,
+} from './consumer'
+
+export const startBatchConsumer = async ({
+    connectionConfig,
+    groupId,
+    topic,
+    sessionTimeout,
+    consumerMaxBytesPerPartition,
+    consumerMaxBytes,
+    consumerMaxWaitMs,
+    fetchBatchSize,
+    eachBatch,
+}: {
+    connectionConfig: GlobalConfig
+    groupId: string
+    topic: string
+    sessionTimeout: number
+    consumerMaxBytesPerPartition: number
+    consumerMaxBytes: number
+    consumerMaxWaitMs: number
+    fetchBatchSize: number
+    eachBatch: (messages: Message[]) => Promise<void>
+}) => {
+    // Starts consuming from `topic` in batches of `fetchBatchSize` messages,
+    // with consumer group id `groupId`. We use `connectionConfig` to connect
+    // to Kafka. We commit offsets after each batch has been processed,
+    // disabling the default auto commit behaviour.
+    //
+    // The general purpose of processing in batches is that it allows e.g. some
+    // optimisations to be made to database queries, or batching production to
+    // Kafka.
+    //
+    // Note that we do not handle any pre-fetching explicitly, rather
+    // node-rdkafka will fill it's own internal queue of messages as fast as it
```

Comment:
question: how big is this queue? We're using default values here, right?

Reply:
Yes have added these explicitly to the consumer config below.

Reply:
1GB looks indeed too big for this workload, safer values would be: 
- 'queued.min.messages': 5000 (that's 10 batches) 
- 'queued.max.messages.kbytes': 256000, (that's 250MiB)